src/UnigramWordPredictor.java

```diff
@@ -101,7 +146,38 @@ public void train(Scanner scanner) {
   public String predictNextWord(List<String> context) {
     // TODO: Return a predicted word given the words preceding it
     // Hint: only the last word in context should be looked at
-    return null;
+
+    // check if context is empty 
+    if (context.isEmpty()){
+      return null;
+    }
+
+    // grabbing the checkedWord
+    String checkedWord = context.get(context.size() - 1);
+
+
+
+    if(getNeighborMap().containsKey(checkedWord)){
+
+      List<String> neighbor = getNeighborMap().get(checkedWord);
+
+    if(neighbor != null){
+      // got this from googling
+
+      // essentially threadlocalrandom is used
+      // to run mutliple threadlocalrandoms to generte
+      // numbes randomly whcih would allow you to protect #s randomly
+
+      // just have to ensure that you put the current method on it to let the class know to make its own random number on this instance similar to this. 
+
+      // nextInt() is there to generate a random number based on nthe size of the neighbor.size()
+      int randomInt = ThreadLocalRandom.current().nextInt(neighbor.size());
+//cant use nextInt() so I gotta use math.rnadom instead
+      return neighbor.get(randomInt);
+    }
+    }
+
+    return "I have no idea how we got here!";
```

Comment:
I think we would get here if we were passed a word in our context that we had never been trained on, or that was only the last word in our training data. In a case like this, consider returning null or raising an exception instead.